src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationTestFixture.cs

```diff
@@ -12,20 +14,28 @@ namespace Microsoft.AspNetCore.Mvc.Testing
     /// <summary>
     /// Fixture for bootstrapping an application in memory for functional end to end tests.
     /// </summary>
+    /// <typeparam name="TTestClass">The test class type.</typeparam>
     /// <typeparam name="TStartup">The applications startup class.</typeparam>
-    public class WebApplicationTestFixture<TStartup> : IDisposable where TStartup : class
+    public class WebApplicationTestFixture<TTestClass, TStartup> : IDisposable where TTestClass : class where TStartup : class
```

Comment:
It would be nice if we find a better name for this class.
WebApplicationFactory
WebApplicationBootstrapper
WebApplicationSetup